src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -1641,23 +1641,34 @@ void C2_MacroAssembler::load_vector_mask(KRegister dst, XMMRegister src, XMMRegi
   }
 }
 
-void C2_MacroAssembler::load_vector(XMMRegister dst, Address src, int vlen_in_bytes) {
-  switch (vlen_in_bytes) {
-  case 4:  movdl(dst, src);   break;
-  case 8:  movq(dst, src);    break;
-  case 16: movdqu(dst, src);  break;
-  case 32: vmovdqu(dst, src); break;
-  case 64: evmovdquq(dst, src, Assembler::AVX_512bit); break;
-  default: ShouldNotReachHere();
+void C2_MacroAssembler::load_vector(BasicType bt, XMMRegister dst, Address src, int vlen_in_bytes) {
+  if (is_integral_type(bt)) {
+    switch (vlen_in_bytes) {
+      case 4:  movdl(dst, src);   break;
+      case 8:  movq(dst, src);    break;
+      case 16: movdqu(dst, src);  break;
+      case 32: vmovdqu(dst, src); break;
+      case 64: evmovdqul(dst, src, Assembler::AVX_512bit); break;
+      default: ShouldNotReachHere();
+    }
+  } else {
+    switch (vlen_in_bytes) {
+      case 4:  movflt(dst, src); break;
+      case 8:  movdbl(dst, src); break;
+      case 16: movups(dst, src); break;
+      case 32: vmovups(dst, src, Assembler::AVX_256bit); break;
+      case 64: vmovups(dst, src, Assembler::AVX_512bit); break;
+      default: ShouldNotReachHere();
+    }
```

Comment:
Vector Load/store from memory happens from dedicated ports, can you elaborate why this change will benefit.

Reply:
> Vector Load/store from memory happens from dedicated ports, can you elaborate why this change will benefit.

Above reference to section 3.5.5.2 also states that FP loads adds another cycle of latency,  but saving the cycles penalty due to bypass b/w FP and SIMD domains still holds good. So may be for load there is no pressing need and existing load vector handling can be kept as it is.

Overall savings from constant table size reductions are very impressive.  Thanks.

Reply:
Thanks for your sharing, I have reverted the change here